### PR TITLE
add file filters to ScoreLibrary

### PR DESCRIPTION
**Pull request by nittka**
This PR introduces some filters allowing to hide files that may not be of interest at some point.

### CODE REVIEW

**Comment by nittka** [comment]
The second commit adds the navigator filter for closed projects and excludes closed projects from the Non-LilyPond projects.
Previously, you could re-open a closed project only by deactivating the Non-LilyPond filter - even if the closed project was a LilyPond project. Separating these two filters seems cleaner.

File: org.elysium.parent/org.elysium.ui/plugin.xml [hunk 1]
```diff
@@ -182,9 +182,15 @@
       <includes>
         <contentExtension pattern="org.eclipse.ui.navigator.resourceContent"/>
         <contentExtension pattern="org.eclipse.ui.navigator.resources.filters.startsWithDot"/>
+        <contentExtension pattern="org.eclipse.ui.navigator.resources.filters.closedProject"/>
         <contentExtension pattern="org.eclipse.ui.navigator.resources.linkHelper"/>
         <contentExtension pattern="org.eclipse.ui.navigator.resources.workingSets"/>
         <contentExtension pattern="org.elysium.ui.filters.NonLilyPondProjects"/>
+        <contentExtension pattern="org.elysium.ui.filters.midi"/>
```
Comment (thSoft) [inline org.elysium.parent/org.elysium.ui/plugin.xml]: I don't clearly see the use case of hiding LilyPond or compiled (midi, pdf, ps) files from the Score Library, which is exactly for browsing these kinds of files. Would you mind keeping only the tempfile filter?

Comment (nittka) [reply org.elysium.parent/org.elysium.ui/plugin.xml]: My intention for hiding the compiled files is the following: Opening them is possible also via "Show in" on the sources. I generally like a "clean" workspace, i.e. seeing only what I need. So hiding the compile results (potentially numerous) allows to concentrate on the sources to edit (also potentially numerous).

The idea of being able to hide the source is the reverse. If I want to see which scores I have, showing only the pdfs gives a faster overview.

Actually, I'd like to keep all the filters. It's not like you *have* to use them. But to make sure your question about the filters to leave out, could you explicitly enumerate those you'd keep?

**Comment by nittka** [comment]
Please review my response, note that the by default the controversial filters are not active.

**Comment by thSoft** [comment]
Note that the "Sort By Extension" sorter exists for the purpose to distinguish the source files from the compiled files easily. But since these filters are disabled by default, they do not clutter the default workflow, and nothing depends on them, it's OK to have them, in case you want to filter source or compiled files.

**Pull request merged**